exercises/alpha_beta/descr.md

```diff
@@ -0,0 +1,501 @@
+on demand, and can be reclaimed by the garbage collector when they are no
+longer needed. Thus, at a given moment, only one branch, namely the branch
+that leads from the root of the tree down to the tree node under examination,
+actually needs to reside in memory.
+
+In this relatively simple-minded approach, there is *no sharing* of subtrees.
+If two moves lead to the same game state, then the game tree contains two
+identical subtrees. There is unfortunately no way of recognizing that these
+subtrees are identical and (thereby) to avoid duplicated work. It is possible
+to do so, using a hash table whose keys are game states, but this is not done
+in this exercise.
+
+## On-Demand Sequences
+
+The type of **on-demand sequences** is defined in a module named `Seq`.
+Beginning with version 4.07,
+this module is part of OCaml's standard library.
+
+```
+module Seq : sig
+
+  type 'a t = unit -> 'a node
+
+  and +'a node =
+  | Nil
+  | Cons of 'a * 'a t
+
+end
+```
+
+This data type is closely related to the algebraic data type of lists.
+Indeed, if instead of `unit -> 'a node` one had written just `'a node`,
+then this data type would have been isomorphic to the type of lists.
+
+The presence of `unit -> ...` indicates that a sequence is in fact a function.
+Calling this function, by applying it to the value `()`, amounts to requesting
+the head of the sequence. This head can be either `Nil`, which means that the
+sequence is empty, or `Cons (x, xs)`, which means that the first element of
+the sequence is `x` and the remaining elements form another sequence `xs`. It
+is worth noting that `xs` is itself a function, so the elements of the
+sequence `xs` need not be explicitly computed until `xs` is applied.
+
+## Warmup: Size and Height
+
+As a warmup, let us implement two functions that compute the size and the
+height of a game tree.
+
+The **size** of a game tree is the number of its nodes (including both leaf
+and nonleaf nodes).
+
+The **height** of a game tree is the maximum length of a branch. In other
+words, it is the maximum number of moves that can be played consecutively,
+beginning in the initial game state.
+
+**Question 1.** Define the function `size`,
+whose type is `tree -> int`.
+
+**Question 2.** Define the function `height`,
+whose type is `tree -> int`.
+
+## Evaluating a Game Tree: the Minimax Algorithm
+
+Suppose that we are given a game tree. We wish to **evaluate** it, that is, to
+compute the best possible outcome of the game for the player who is initially
+up (the one who plays the first move), under the assumption that his opponent
+plays perfectly. We refer to this player as `Even`, and refer to his opponent
+as `Odd`. This is reflected in the following type definition:
+
+```
+type sense =
+  | Even
+  | Odd
+```
+
+The following auxiliary functions will be useful below. In particular,
+`interpret sense` transforms a value that is relative to the player `sense`
```

Comment:
```suggestion
`interpret sense` transforms a value that is relative to the player `Odd`
```

Reply:
Actually, I am not sure about this one. I guess you want to use Odd and Even the other way around here.

Reply:
It seems to me that the text is correct.